google/bigtable_client_factory.go

```diff
@@ -0,0 +1,22 @@
+package google
+
+import (
+	"context"
+
+	"cloud.google.com/go/bigtable"
+	"golang.org/x/oauth2"
+	"google.golang.org/api/option"
+)
+
+type BigtableClientFactory struct {
+	UserAgent   string
+	TokenSource oauth2.TokenSource
+}
+
+func (s *BigtableClientFactory) NewInstanceAdminClient(project string) (*bigtable.InstanceAdminClient, error) {
```

Comment:
If you remove the pointer on the receiver in the signature (e.g. 
```
func (s BigtableClientFactory) NewInstanceAdminClient(project string) (*bigtable.InstanceAdminClient, error) {
```
), then this function is much more explicit that it is immutable (and is therefore thread safe, etc)